src/krylov/gmres.c

```diff
@@ -361,7 +376,8 @@ hypre_GMRESSolve(void  *gmres_vdata,
    /* compute initial residual */
    (*(gmres_functions->Matvec))(matvec_data, -1.0, A, x, 1.0, p[0]);
 
-   b_norm = hypre_sqrt((*(gmres_functions->InnerProd))(b, b));
+   (*(gmres_functions->InnerProd))(b, b, &num_tags, &biprod);
+   b_norm = hypre_sqrt(biprod[0]);
```

Comment:
Maybe I am misreading the implementation of InnerProdTagged() but it looks like only a subset of values in b will be involved in the inner product.  If everything is tagged with a nonzero value, then b_norm would be zero right?  Is the intention to converge based on tag zero?

Reply:
The first entry in `biprod` will be the full inner prod (considering all the tags). So this is backward compatible. Here's a comment about the meaning of the entries:

https://github.com/hypre-space/hypre/blob/70f3970ba5e57f7167ef4a3c15349b5d19618de5/src/parcsr_mv/par_vector.c#L615-L617